ironic/drivers/modules/drac/management.py

```
@@ -302,14 +315,182 @@ def set_boot_device(node, device, persistent=False):
+        for manager in system.managers:
+            try:
+                oem_manager = manager.get_oem_extension('Dell')
+            except sushy.exceptions.OEMExtensionNotFoundError as e:
+                error_msg = (_("Search for Sushy OEM extension package "
+                           "'sushy-oem-idrac' failed for node %(node)s. "
+                           "Ensure it is installed. Error: %(error)s") %
+                         {'node': task.node.uuid, 'error': e})
+                LOG.error(error_msg)
+                raise exception.RedfishError(error=error_msg)
+            try:
+                reset_job_response = oem_manager.reset_idrac(
+                                                manager=manager)
+            except sushy.exceptions.SushyError as e:
+                LOG.debug("Sushy OEM extension Python package "
+                          "sushy-oem-idrac failed to reset idrac"
+                          "for %(node)s, Error : %(error)s" %
+                          {'node':task.node.uuid, 'error':e})
+        url = oem_manager._conn._url
+        redfish_node_ip = re.search(r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}',
+                                    url).group()
+        if reset_job_response.status_code == _RESET_JOB_RESPONSE_CODE:
+            LOG.info("idrac reset success for node %(node)s via OEM" %
+                    {'node': task.node.uuid})
+            LOG.debug("iDRAC was reset, waiting for return to operational state")
+            redfish_utils.wait_for_host(redfish_node_ip)
+            LOG.info("The iDRAC has become pingable")
+            LOG.info("Waiting for the iDRAC to become ready")
+            redfish_utils.wait_until_idrac_is_ready(oem_manager,
+                        retries=IDRAC_IS_READY_RETRIES,
+                        retry_delay=IDRAC_IS_READY_RETRY_DELAY_SEC)
+        else:
+            LOG.error("Failed to reset idrac for %(node)s" %
+                    {'node': task.node.uuid})
```

We should raise an exception here so Ironic knows cleaning failed.

---

Addressed, raised an exception on failure.